framework/src/media/audio/audio_manager.c

```diff
@@ -1299,25 +1350,281 @@ audio_manager_result_t find_stream_in_device_with_process_type(device_process_ty
 					pthread_mutex_unlock(&(card->card_mutex));
 					continue;
 				}
+
+				subprocess_type = get_subprocess_type_audio_param_value(subtype);
+
 				/* Now we get subtype from h/w, compare it and subtype(param) */
-				if ((subtype & caps_desc.caps.ac_controls.b[0]) != 0) {
+				if ((subprocess_type & caps_desc.caps.ac_controls.b[0]) != 0) {
 					*card_id = i;
 					*device_id = j;
 					close(fd);
 					pthread_mutex_unlock(&(card->card_mutex));
 					return AUDIO_MANAGER_SUCCESS;
 				}
+				pthread_mutex_unlock(&(card->card_mutex));
 			}
 		}
 	}
 
-	meddbg("We didn't find any device with process type : %d subtype : %d\n", type, subtype);
+	meddbg("We didn't find any device with process type : %d subtype : %d process_type : %d subprocess_type : %d\n", type, subtype, process_type, subprocess_type);
 	return AUDIO_MANAGER_DEVICE_NOT_SUPPORT;
 }
 
+audio_manager_result_t request_stream_in_device_process(int card_id, int device_id, int cmd)
+{
+	char path[AUDIO_DEVICE_FULL_PATH_LENGTH];
+	int fd;
+	audio_card_info_t *card;
+	audio_config_t *config;
+	int ret;
+
+	card = &g_audio_in_cards[card_id];
+	config = &card->config[device_id];
+
+	/* Check card register state first */
+	if (config->process_handler == NULL) {
+		return AUDIO_MANAGER_CARD_NOT_READY;
+	}
+
+	get_card_path(path, card_id, device_id, INPUT);
+	pthread_mutex_lock(&(card->card_mutex));
+	fd = open(path, O_RDONLY);
+	if (fd < 0) {
+		meddbg("Failed to open audio driver, path : %s errno : %d\n", path, errno);
+		pthread_mutex_unlock(&(card->card_mutex));
+		return AUDIO_MANAGER_NO_AVAIL_CARD;
+	}
+
+	ret = ioctl(fd, cmd, 0);
+	if (ret < 0) {
+		meddbg("Failed to start process ret : %d\n", ret);
+		pthread_mutex_unlock(&(card->card_mutex));
+		return AUDIO_MANAGER_DEVICE_FAIL;
```

Comment:
Doesn't fd(close) need to be added before this return?

Reply:
I will fix it.